internal/app/config.go

```diff
@@ -76,50 +90,63 @@ func loadConfig(path string) (*Configuration, error) {
 	return cfg, nil
 }
 
-func validateClientParams(cfg *Configuration) error {
-	errCfgInvalid := errors.New("Configuration is invalid")
+func loadEnvOverrides(cfg *Configuration, v *viper.Viper) error {
+	cfg.FdbCfg.ClientSecret = v.GetString("fleetdb.oidc.client.secret")
+	if cfg.FdbCfg.ClientSecret == "" {
+		return errors.New("FLEET_SCHEDULER_FLEETDB_OIDC_CLIENT_SECRET was empty")
+	}
+
+	cfg.CoCfg.ClientSecret = v.GetString("conditionorc.oidc.client.secret")
+	if cfg.FdbCfg.ClientSecret == "" {
+		return errors.New("FLEET_SCHEDULER_CONDITIONORC_OIDC_CLIENT_SECRET was empty")
+	}
+
+	return nil
+}
 
+func validateClientParams(cfg *Configuration) error {
 	if cfg.LogLevel == "" {
-		cfg.LogLevel = LogLevelInfo
-	} else {
-		if cfg.LogLevel != LogLevelInfo &&
-			cfg.LogLevel != LogLevelDebug &&
-			cfg.LogLevel != LogLevelTrace {
-				return errors.Wrap(errCfgInvalid, "LogLevel")
-			}
+		cfg.LogLevel = "debug"
+	}
+
+	if cfg.Concurrency <= 0 {
+		cfg.Concurrency = defaultConcurrencyCount
 	}
 
 	// FleetDB (serverservice) Configuration
 	if cfg.FdbCfg == nil {
-		return errors.Wrap(errCfgInvalid, "fleetdb_api entry doesnt exist")
+		return errors.Wrap(ErrInvalidConfig, "fleetdb_api entry doesnt exist")
 	}
 	if cfg.CoCfg == nil {
-		return errors.Wrap(errCfgInvalid, "conditionorc_api entry doesnt exist")
+		return errors.Wrap(ErrInvalidConfig, "conditionorc_api entry doesnt exist")
 	}
 	if cfg.FacilityCode == "" {
-		return errors.Wrap(errCfgInvalid, "Facility Code")
+		return errors.Wrap(ErrInvalidConfig, "Facility Code")
 	}
-	err := validateOIDCConfig(cfg.FdbCfg, errors.Wrap(errCfgInvalid, "fleetdb_api is invalid"))
+
+	err := validateOIDCConfig(cfg.FdbCfg, defaultFleetDBClientID)
 	if err != nil {
 		return err
 	}
-	err = validateOIDCConfig(cfg.CoCfg, errors.Wrap(errCfgInvalid, "conditionorc_api is invalid"))
+	err = validateOIDCConfig(cfg.CoCfg, defaultConditionOrcClientID)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func validateOIDCConfig(cfg* ConfigOIDC, err error) error {
+func validateOIDCConfig(cfg *ConfigOIDC, defaultClientID string) error {
+	if cfg.ClientID == "" {
```

Comment:
this conditional might be nicer up a level.

Reply:
its specific to fleetdb or conditionorc